service/worker/service.go

```diff
@@ -280,8 +300,7 @@ func (svc *Service) lead(ctx context.Context) error {
 		WatchPath: "metro/subscriptions",
 		Handler: func(ctx context.Context, pairs []registry.Pair) {
 			logger.Ctx(ctx).Infow("nodes watch handler data", "pairs", pairs)
-			newSubs := registry.GetKeys(pairs)
-			svc.handleSubUpdates(newSubs)
+			svc.handleSubUpdates(pairs)
```

Comment:
`handleNodeUpdates ` and `handleSubUpdates ` happens concurrently. Do you see any side-effects? Does this need to be synchronized?

Reply:
concurrently is fine, issue is with the failures in both cases, as watch is triggered only once, we need some intermediate storage if processing fails

i.e. if a subscription scheduling fails, put that in some channel and process it in separate go routine